src/lambda/events/generate_test_event.py

```diff
@@ -42,81 +45,111 @@
 }
 
-def create_message_template(project_id, folder_id):
+def create_message_template(entity):
   return {
     "appId": "example-app-1",
-    "recordId": "-4I2GOqDSdjaXsbuw8oYXBKK",
+    "recordId": entity.annotations.recordId.pop(),
     "record": {
-      "parentProjectId": project_id,
-      "rawFolderId": folder_id,
-      "fileEntityId": "",
-      "s3Bucket": "",
-      "s3Key": ""
+      "parentProjectId": "syn0000000",
+      "rawFolderId": entity.parentId,
+      "fileEntityId": entity.id,
+      "s3Bucket": entity._file_handle.bucketName,
+      "s3Key": entity._file_handle.key
     },
     "studyRecords": {
       "study-1": {
-        "parentProjectId": project_id,
-        "rawFolderId": folder_id,
-        "fileEntityId": "",
-        "s3Bucket": "",
-        "s3Key": ""
+        "parentProjectId": "syn0000000",
+        "rawFolderId": "syn0000000",
+        "fileEntityId": "syn0000000",
+        "s3Bucket": "dummy-bucket",
+        "s3Key": "dummy-key"
       }
     }
   }
 
 
 def read_args():
   parser = argparse.ArgumentParser(
-    description='Generate a json file of a mocked SNS event for testing.')
+    description='Generate a json file of a mocked SQS event for testing.')
   parser.add_argument('--synapse-project-id',
     default='syn26721259',
     help='Synapse ID of the BridgeDownstreamTest project')
+  parser.add_argument('--dataset-name',
+    default='bridge-downstream-test-dataset',
+    help=('Optional. Name of the Synapse dataset containing test data. '
+          'Default is "bridge-downstream-test-dataset".'))
   args = parser.parse_args()
   return args
 
 
-def main():
-  args = read_args()
-  project_id = args.synapse_project_id
-
+def get_synapse_client(ssm_parameter):
+  '''Get an instance of the synapse client'''
+  ssm_client = boto3.client('ssm')
+  token = ssm_client.get_parameter(
+    Name=ssm_parameter,
+    WithDecryption=True)
   syn = synapseclient.Synapse()
-  syn.login()
-
-  folder_name = 'test-data'
-  print(f'Get folder id')
-  response = list(syn.getChildren(project_id, includeTypes=['folder']))
-  folder = next(item for item in response if item['name'] == folder_name)
-  folder_id = '' if folder is None else folder.get('id')
-  if not folder_id:
-    print(f'No folder {folder_name} exists')
+  synapse_auth_token = token['Parameter']['Value']
+  syn.login(authToken=synapse_auth_token, silent=True)
+  return syn
+
+
+def get_dataset_id_by_name(syn, parent_id, dataset_name):
+  response = list(syn.getChildren(parent_id, includeTypes=['dataset']))
+  dataset = next(item for item in response if item['name'] == dataset_name)
+  dataset_id = '' if dataset is None else dataset.get('id')
+  if not dataset_id:
+    print(f'No dataset {dataset_name} exists')
     sys.exit(1)
+  return dataset_id
+  return dataset
```

Comment:
Two return statements?

Reply:
Whoops. I'll lint my changes and commit.